filclient/filc/main.go

```diff
@@ -433,20 +462,162 @@ var queryRetrievalCmd = &cli.Command{
+	Usage:     "Delete all retrieved file data in the blockstore",
+	ArgsUsage: " ",
+	Action: func(cctx *cli.Context) error {
+		ddir := ddir(cctx)
+
+		fmt.Println("clearing blockstore...")
+
+		if err := os.RemoveAll(blockstorePath(ddir)); err != nil {
+			return err
 		}
 
+		fmt.Println("done")
+
 		return nil
 	},
 }
+
+func printAskResponse(ask *storagemarket.StorageAsk) {
+	fmt.Printf(`ASK RESPONSE
+-----
+Miner: %v
+Price (Unverified): %v (%v)
+Price (Verified): %v (%v)
+Min Piece Size: %v
+Max Piece Size: %v
+`,
+		ask.Miner,
+		ask.Price, types.FIL(ask.Price),
+		ask.VerifiedPrice, types.FIL(ask.VerifiedPrice),
+		ask.MinPieceSize,
+		ask.MaxPieceSize,
+	)
+}
+
+func printRetrievalStats(stats *filclient.RetrievalStats) {
+	fmt.Printf(`RETRIEVAL STATS
+-----
+Size:          %v (%v)
+Total Payment: %v (%v)
+Ask Price:     %v (%v)
+Num Payments:  %v
+Duration:      %v
+Average Speed: %v (%v/s)
+Peer:          %v
+`,
+		stats.Size, formatBytes(stats.Size),
+		stats.TotalPayment, types.FIL(stats.TotalPayment),
+		stats.AskPrice, types.FIL(stats.AskPrice),
+		stats.NumPayments,
+		stats.Duration,
+		stats.AverageSpeed, formatBytes(stats.AverageSpeed),
+		stats.Peer,
+	)
+}
+
+func printQueryResponse(query *retrievalmarket.QueryResponse) {
+	var status string
+	switch query.Status {
+	case retrievalmarket.QueryResponseAvailable:
+		status = "Available"
+	case retrievalmarket.QueryResponseUnavailable:
+		status = "Unavailable"
+	case retrievalmarket.QueryResponseError:
+		status = "Error"
+	default:
+		status = fmt.Sprintf("Unrecognized Status (%d)", query.Status)
+	}
+
+	var pieceCIDFound string
+	switch query.PieceCIDFound {
+	case retrievalmarket.QueryItemAvailable:
+		pieceCIDFound = "Available"
+	case retrievalmarket.QueryItemUnavailable:
+		pieceCIDFound = "Unavailable"
+	case retrievalmarket.QueryItemUnknown:
+		pieceCIDFound = "Unknown"
+	default:
+		pieceCIDFound = fmt.Sprintf("Unrecognized (%d)", query.PieceCIDFound)
+	}
+
+	total := big.Add(query.UnsealPrice, big.Mul(big.NewIntUnsigned(query.Size), query.MinPricePerByte))
+	fmt.Printf(`QUERY RESPONSE
+-----
+Status:                        %v
+Piece CID Found:               %v
+Size:                          %v (%v)
+Unseal Price:                  %v (%v)
+Min Price Per Byte:            %v (%v)
+Total Retrieval Price:         %v (%v)
+Payment Address:               %v
+Max Payment Interval:          %v (%v)
+Max Payment Interval Increase: %v (%v)
+`,
+		status,
+		pieceCIDFound,
+		query.Size, formatBytes(query.Size),
+		query.UnsealPrice, types.FIL(query.UnsealPrice),
+		query.MinPricePerByte, types.FIL(query.MinPricePerByte),
+		total, types.FIL(total),
+		query.PaymentAddress,
+		query.MaxPaymentInterval, formatBytes(query.MaxPaymentInterval),
+		query.MaxPaymentIntervalIncrease, formatBytes(query.MaxPaymentIntervalIncrease),
+	)
+
+	if query.Message != "" {
+		fmt.Printf("Message: %v\n", query.Message)
+	}
+}
+
+func formatBytes(count uint64) string {
```

Comment:
Heh, I appreciate the effort, but you can also just use: https://github.com/dustin/go-humanize

Reply:
oo, looks much better than my scrap function LOL